doc/developer-guides/hld/hv-rdt.rst

```diff
@@ -0,0 +1,54 @@
+.. _hv_rdt:
+
+RDT Allocation Feature Supported by Hypervisor
+##############################################
+
+The hypervisor allows to use RDT (Resource Director Technology) allocation features to optimize performances of VMs. There are 2 sub-features: CAT (Cache Allocation Technology) and MBA(Memory Bandwidth Allocation), CAT is for cache resources and MBA is for memory bandwidth resources. The RDT in hypervisor is integrated with VM-Configuration feature, the resources consumed by each VM can be configured and restricted at compile time.
+
+CAT High-Level Design
+*********************
+
+Introduction of CAT capabilities
+===============================
+
+On a platform which support CAT, each CPU can mask last-level-cache (LLC) with a cache mask, the masked cache ways cannot be evicted by this CPU. In terms of SDM, CAT abilities are enumerated via CPUID, and configured via MSR registers, these are:
+
+* CPUID.10H contains CAT capabilities, such as cache mask type(L2/L3), mask bit
+  length, number of masks.
+
+* Cache masks are set aside in IA32_type_MASK_n, each one of these MSRs can
+  hold a cache mask. They are shared by CPUs who share the same LLC; CPU set
+  the class-of-service (CLOS) of IA32_type_MASK_n in its IA32_PQR_ASSOC MSR, to
+  enable that cache mask.
+
+Objective of CAT
+================
+
+CAT feature in hypervisor can isolate cache for a VM from other VMs. It can also isolate a VM from hypervisor. So that the VM has its dedicated cache, is running without much unexpected delays for fetching instruction & data.
+If CAT is supported by the platform, it is enabled by default. CPUs are sharing their entire LLC. To change that, the cache masks array, CLOS for VM guest state and host state can be assigned to another values, within these places:
+
+* board.c, the board specified configuration source file. The values of MSRs
+  IA32_type_MASK_n should be defined here.
+* clos, the CLOS of a VM, in its vm_config data structure. The vm_config
+  data structures are in the scenario specified configuration source file.
+* hv_clos, the CLOS of hypervisor. It is in hypervisor CAT source code.
```

Comment:
These are not necessary for HLD, could be removed.

Reply:
removed